superset-frontend/packages/superset-ui-core/src/time-comparison/getTimeOffset.ts

```diff
@@ -78,6 +78,11 @@ export const parseDttmToDate = (
         now.setMonth(now.getMonth() - 1);
       }
       return now;
+    case 'previous calendar quarter': {
```

Comment:
@personofnorank You can add tests to the` getTimeOffset.test.ts` file.

Reply:
Done :)

Reply:
Hi @personofnorank. There are problems with this logic as it can result in an invalid month value (e.g., -3, -2, -1) when the current quarter is the first quarter of the year.

I suggest adding some tests like these:

```
test('should return the first day of the previous calendar quarter when current date is in Q4', () => {
  const result = getTimeOffset('previous calendar quarter', false, now);
  expect(result).toEqual(new Date('2023-07-01T00:00:00Z'));
});

test('should return the first day of the previous calendar quarter when current date is in Q3', () => {
  now = new Date('2023-08-15T00:00:00Z');
  const result = getTimeOffset('previous calendar quarter', false, now);
  expect(result).toEqual(new Date('2023-04-01T00:00:00Z'));
});

test('should return the first day of the previous calendar quarter when current date is in Q2', () => {
  now = new Date('2023-05-15T00:00:00Z');
  const result = getTimeOffset('previous calendar quarter', false, now);
  expect(result).toEqual(new Date('2023-01-01T00:00:00Z'));
});

test('should return the first day of the previous calendar quarter when current date is in Q1', () => {
  now = new Date('2023-02-15T00:00:00Z');
  const result = getTimeOffset('previous calendar quarter', false, now);
  expect(result).toEqual(new Date('2022-10-01T00:00:00Z'));
});
```

Reply:
It should work because Date allows a negative month.
Date(2016, -1) gives you December 2015.